### DISCUSSION TITLE
Expected "java: Several possible source properties for target property "name"." when using String as parameter

### DISCUSSION THREAD

**Discussion by jvwilge** [mapstruct/mapstruct#3242] [General]
I'm not sure whether this is an issue (or just **my** issue ;-) )but the following example illustrates what's happening :

```java
import lombok.Builder;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;
import org.mapstruct.Mapper;

@Mapper
public interface HouseMapper {

    // java: Several possible source properties for target property "name".
    HouseDto toHouseDto(House house, House house2);

    // Gives no error, but String name is ignored
    HouseDto toHouseDto(House house, String name);
}

@Value
@Jacksonized
@Builder(toBuilder = true)
class House {
    String name;
}

@Value
@Jacksonized
@Builder(toBuilder = true)
class HouseDto {
    String name;
}
```

The first `toHouseDto` shows the error : `java: Several possible source properties for target property "name".` which is expected since there are 2 `name` candidates.

I expected the same message on the second `toHouseDto` since this also has multiple options. There is no error and code is generated with the `name` from the second parameter ignored.

This all was run in a simple maven project without any mapstruct options in the annotation processor.

Thanks for your help!

**Comment by filiphr** [comment]
If you read https://mapstruct.org/documentation/stable/reference/html/#mappings-with-several-source-parameters especially part:

>In case several source objects define a property with the same name, the source parameter from which to retrieve the property must be specified using the `@Mapping` annotation as shown for the description property in the example. An error will be raised when such an ambiguity is not resolved. For properties which only exist once in the given source objects it is optional to specify the source parameter’s name as it can be determined automatically.

This means that the source parameters are treated as objects. i.e. the several possible source properties is only shown if multiple parameters contain a property with the same name. In the second example `String` does not have a property names `name`. That's why you do not have an error.
